rskj-core/src/main/java/co/rsk/net/discovery/message/NeighborsPeerMessage.java

```diff
@@ -105,6 +85,28 @@ public static NeighborsPeerMessage create(List<Node> nodes, String check, ECKey
         return neighborsMessage;
     }
 
+    @Override
+    protected final void parse(byte[] data) {
+        RLPList list = (RLPList) RLP.decode2OneItem(data, 0);
+
+        if (list.size() < 2) {
+            throw new PeerDiscoveryException(MORE_DATA);
+        }
+
+        RLPList nodesRLP = (RLPList) list.get(0);
+
+        for (int i = 0; i < nodesRLP.size(); ++i) {
+            RLPList nodeRLP = (RLPList) nodesRLP.get(i);
+            Node node = new Node(nodeRLP.getRLPData());
+            nodes.add(node);
+        }
+
+        RLPItem chk = (RLPItem) list.get(1);
+        this.messageId = extractMessageId(chk);
+
+        this.setNetworkIdWithRLP(list.size() > 2 ? list.get(2) : null);
+    }
+
```

Comment:
I've just moved down this block, no new code but for `this.messageId = extractMessageId(chk);`